react-validated-input-component/my-app/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+import { FaCheck } from 'react-icons/fa';
+import { ImCross } from 'react-icons/im';
+
+export function ValidatedInput() {
+  const [password, setPassword] = useState('');
+
+  const length = password.length;
+
+  const includeDigit = /[0-9]/.test(password);
+
+  const includeCapital = /[A-Z]/g.test(password);
+
+  function updateError() {
+    if (
+      length === 0 ||
+      length < 8 ||
+      (length >= 8 && !includeDigit) ||
+      (length >= 8 && !includeCapital)
+    ) {
+      return <ImCross className="invalid" />;
+    } else {
+      return <FaCheck className="valid" />;
+    }
+  }
+
+  function updateTextColor() {
+    if (
+      length === 0 ||
+      length < 8 ||
+      (length >= 8 && !includeDigit) ||
+      (length >= 8 && !includeCapital)
```

Comment:
This is exactly the same check as above, and similar to the one below. Can you figure out how to get rid of this duplication? I think you should be able to do all the testing just once.